Cargo.toml

```diff
@@ -255,6 +255,7 @@ enum_dispatch = { version = "0.3.12", default-features = false }
 exitcode = { version = "1.1.2", default-features = false }
 flate2 = { version = "1.0.28", default-features = false, features = ["default"] }
 futures-util = { version = "0.3.29", default-features = false }
+fs4 = { version = "0.7.0" }
```

Comment:
Looking at the commit message, can you elaborate on how `fslock` didn't work? Do we need to update the file locking logic in the disk buffers to use `fs4` as well (in a separate PR if desired)?